utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -383,6 +420,244 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
     pub fn get_u32(&self, code_point: u32) -> u32 {
         self.get(code_point).into()
     }
+
+    /// Returns a [`CodePointMapRange`] struct representing a range of code
+    /// points associated with the same value.
+    pub fn get_range(&self, start: u32) -> Option<CodePointMapRange> {
+        if CODE_POINT_MAX < start {
+            return None;
+        }
+        if start >= self.header.high_start {
+            let di: usize = self.data.len() - (HIGH_VALUE_NEG_DATA_OFFSET as usize);
+            let value: u32 = if let Some(v) = self.data.get(di) {
+                v.into()
+            } else {
+                return None;
+            };
+            return Some(CodePointMapRange {
+                start,
+                end: CODE_POINT_MAX,
+                value,
+            });
+        }
+
+        let null_value: u32 = self.header.null_value;
+
+        let mut prev_i3_block: u32 = u32::MAX; // using u32::MAX (instead of -1 as an i32 in ICU)
+        let mut prev_block: u32 = u32::MAX; // using u32::MAX (instead of -1 as an i32 in ICU)
+        let mut c: u32 = start;
+        let mut trie_value: u32 = 0;
+        let mut value: u32 = 0;
+        let mut have_value: bool = false;
+
+        loop {
```

Comment:
Nit: This is a big chunk of complicated code. It might be nice if we added a short paragraph somewhere in this function to give the reader an intuition for what it's all doing. (My understanding is that we're iterating through the data in the code point trie until we find a code point with a different value than the starting point of the range; I'm hoping for a comment just a touch more detailed than that.)

Reply:
Looks reasonable. I didn't actually compare it with the code :-)